File: rust/src/models/tax_exemption_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(
    Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
)]
pub enum TaxExemptionType {
    #[default]
    #[serde(rename = "REVERSE_CHARGE")]
    ReverseCharge,

    #[serde(rename = "TAX_EXEMPT")]
    TaxExempt,

    #[serde(rename = "NOT_REGISTERED")]
    NotRegistered,
}

impl fmt::Display for TaxExemptionType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::ReverseCharge => "REVERSE_CHARGE",
            Self::TaxExempt => "TAX_EXEMPT",
            Self::NotRegistered => "NOT_REGISTERED",
        };
        f.write_str(value)
    }
}

File: typescript/src/models/taxExemptionType.ts
// this file is @generated

export enum TaxExemptionType {
  ReverseCharge = "REVERSE_CHARGE",
  TaxExempt = "TAX_EXEMPT",
  NotRegistered = "NOT_REGISTERED",
  Export = "EXPORT",
  NoVatTerritory = "NO_VAT_TERRITORY",
}

export const TaxExemptionTypeSerializer = {
  _fromJsonObject(object: any): TaxExemptionType {
    return object;
  },

  _toJsonObject(self: TaxExemptionType): any {
    return self;
  },
};
